### PR TITLE
Bunch of Android Auto enhancements

### PR DESCRIPTION
**Pull request by JBassett**
## Summary

Enhance loading and state syncing.
Prompt for login when appropriate.
Add ability to navigate to entities with locations.
Add icons to main view.
Make sure we only have one Playstore publish running at once.

## Screenshots

Screenshot 2023-01-13 at 11 00 22

Screenshot 2023-01-13 at 11 25 37

Screenshot 2023-01-13 at 11 27 38

## Link to pull request in Documentation repository

N/A but soon...

### CODE REVIEW

File: app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt [hunk 1]
```diff
@@ -38,26 +57,76 @@ class MainVehicleScreen(
             "script" to commonR.string.scripts,
             "switch" to commonR.string.switches,
         )
+
+        private val DOMAIN_TO_ICON: Map<String, IIcon> = mapOf(
```
Comment (jpelgrom) [inline app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt:62]: On Wear this is handled by [creating a placeholder entity](https://github.com/home-assistant/android/blob/220b73d20f53cd73b1fea9e27f62ec638a73516c/wear/src/main/java/io/homeassistant/companion/android/util/CommonFunctions.kt#L33), so we can have all the icon code in one place and match the frontend. Did you consider using this?

File: app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt [hunk 2]
```diff
@@ -38,26 +57,76 @@ class MainVehicleScreen(
...
+                    }
+                    invalidate()
                 }
             }
-            invalidate()
         }
     }
 
     override fun onGetTemplate(): Template {
+        if (isLoggedIn != null && isLoggedIn == false) {
```
Comment (jpelgrom) [inline app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt:109]: `null` != `false`

```suggestion
        if (isLoggedIn == false) {
```

**Comment by JBassett** [comment]
> Looking at the screenshots, the domain icons and navigation entities could use a higher resolution icon, similar to other entity icons. Double it to 48dp?

Updated

File: app/src/full/java/io/homeassistant/companion/android/vehicle/MapVehicleScreen.kt [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            }
+
+        return ListTemplate.Builder().apply {
+            setTitle(carContext.getString(R.string.aa_navigation))
+            setHeaderAction(Action.BACK)
+            if (loading) {
+                setLoading(true)
+            } else {
+                setLoading(false)
+                listBuilder.setNoItemsMessage("No entities with locations found.")
```
Comment (jpelgrom) [inline app/src/full/java/io/homeassistant/companion/android/vehicle/MapVehicleScreen.kt:107]: This should use a string resource

File: app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt (hunk 1 above)
Comment (JBassett) [reply app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt]: Done

File: app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt (hunk 2 above)
Comment (JBassett) [reply app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt]: Done.

File: app/src/full/java/io/homeassistant/companion/android/vehicle/MapVehicleScreen.kt (hunk 3 above)
Comment (JBassett) [reply app/src/full/java/io/homeassistant/companion/android/vehicle/MapVehicleScreen.kt]: Thanks.

File: app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt [hunk 4]
```diff
@@ -39,25 +60,62 @@ class MainVehicleScreen(
...
     init {
         lifecycleScope.launch {
-            integrationRepository.getEntities()?.forEach { entity ->
-                val domain = entity.domain
-                if (domain in SUPPORTED_DOMAINS) {
-                    entities[entity.entityId] = entity
-                    domains.add(domain)
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                while (isLoggedIn != true) {
+                    delay(1000)
```
Comment (jpelgrom) [inline app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt:78]: This always adds a 1 second delay, even if the user is logged in.

File: app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt [hunk 5]
```diff
@@ -67,30 +125,81 @@ class MainVehicleScreen(
...
 
-        // TODO: Add row for zones so we can start navigation?
+        listBuilder.addItem(
+            Row.Builder()
+                .setImage(
+                    CarIcon.Builder(
+                        IconicsDrawable(
+                            carContext,
+                            CommunityMaterial.Icon3.cmd_map_outline
+                        ).toAndroidIconCompat()
```
Comment (jpelgrom) [inline app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt:175]: Missed this one when increasing the sizes, it is still blurry

Comment (JBassett) [reply app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt:176]: Ooops..... Ty.

**Review by jpelgrom — APPROVED** [review]
Android Auto changes look good.

Can't really test the (confusingly named) beta deploy workflow change but it should work.

**Pull request merged**